rts/motoko-rts/src/idl.rs

```diff
@@ -314,6 +319,9 @@ unsafe fn skip_any_vec(buf: *mut Buf, typtbl: *mut *mut u8, t: i32, count: u32)
         // makes no progress. No point in calling it over and over again.
         // (This is easier to detect this way than by analyzing the type table,
         // where we’d have to chase single-field-records.)
+        for _ in 1..count {
```

Comment:
Better might be to measure the cost of the previous skip and lower the limits based on a multiplier of that cost and return (without this faux for_loop). Then we would exit even earlier!